test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodReportGeneratorTest.cs

```diff
@@ -0,0 +1,507 @@
+                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
+                    },
+                    10, 6, 0, 0, 0, 0, 0, 1, 0, true
+                },
+                new object[]
+                {
+                    Enumerable.Range(1, 6).Select(v => v.ToString()),
+                    new[] { "1", "2", "3", "4", "5", "6", "7" },
+                    new int[] { 200, 200, 200, 200, 200, 200, 200 },
+                    new DateTime[]
+                    {
+                        new DateTime(2020, 1, 1, 9, 10, 12, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 13, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 21, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 22, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 23, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
+                    },
+                    10, 6, 0, 0, 0, 0, 0, 0, 1, false
+                }
+            };
+
+        static Option<NetworkStatusTimeline> NetworkStatusTimeline => MockNetworkStatusTimeline.GetMock(new TimeSpan(0, 0, 0, 0, 5)).Result;
+
+        [Fact]
+        public void TestConstructorSuccess()
+        {
+            string expectedSource = "expectedSource";
+            string actualSource = "actualSource";
+            int batchSize = 10;
+            string resultType = "resultType1";
+
+            var mockExpectedStore = new Mock<ISequentialStore<TestOperationResult>>();
+            var expectedResults = new StoreTestResultCollection<TestOperationResult>(mockExpectedStore.Object, batchSize);
+            var mockActualStore = new Mock<ISequentialStore<TestOperationResult>>();
+            var actualResults = new StoreTestResultCollection<TestOperationResult>(mockActualStore.Object, batchSize);
+
+            var reportGenerator = new DirectMethodReportGenerator(
+                Guid.NewGuid().ToString(),
+                expectedSource,
+                expectedResults,
+                actualSource,
+                actualResults,
+                resultType,
+                new DirectMethodTestOperationResultComparer(),
+                NetworkStatusTimeline);
+
+            Assert.Equal(actualSource, reportGenerator.ActualSource);
+            Assert.Equal(actualResults, reportGenerator.ActualTestResults);
+            Assert.Equal(expectedSource, reportGenerator.ExpectedSource);
+            Assert.Equal(expectedResults, reportGenerator.ExpectedTestResults);
+            Assert.Equal(resultType, reportGenerator.ResultType);
+            Assert.Equal(typeof(DirectMethodTestOperationResultComparer), reportGenerator.TestResultComparer.GetType());
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void TestConstructorThrowsWhenTrackingIdIsNotProvided(string trackingId)
```

Comment:
I agree. I'll remove unless anyone thinks otherwise. I was following CountingReportGeneratorTest, so maybe there is a good reason

Reply:
We would like to have test coverage for every single line of code.  This test is verify constructor will throw exception if tracking id is not provided.  There are 2 ways to verify, 1) mock preconditions and verify precondition method get called. 2) Verify that argument exception throws.  We took 2nd option because it is easier to implement.

Reply:
@philipktlin I see where you're coming from. But doesn't that implies that we should also verify the constructor correctly set the correct input argument to the right variable? 

@dylanbronson  Per Phililps' comment, I think it would make sense to do `[InlineData(null)]` and `[InlineData("cash me outside how bout dat")]`? What do you think? I mean we are not really want to test if `Precontidion.*` correctly catch both `null` and `""` - that is done. We want to test more in lue of the constructor if it can handle input argument correctly.

Reply:
@yophilav Do you mean we should test:
1. Null case
2. Case where argument is given correctly?
(and not 3. "")

Currently, we test null, empty string, and we test all the arguments get correctly set in the test named "TestConstructorSuccess." Does that cover what you are asking?

Reply:
yeah.

Reply:
Oh I see. There are 2 cases doing somewhat similar testing. It's fine then.